sonarqube/content.md

```diff
@@ -30,68 +45,146 @@ $ mvn sonar:sonar -Dsonar.host.url=http://$(boot2docker ip):9000
+By default, the image will use an embedded H2 database that is not suited for production.
 
-```console
-sysctl -w vm.max_map_count=262144
-sysctl -w fs.file-max=65536
-ulimit -n 65536
-ulimit -u 4096
-```
+> Warning: Only a single instance of SonarQube can connect to a database schema. If you're using a Docker Swarm or Kubernetes, make sure that multiple SonarQube instances are never running on the same database schema simultaneously. This will cause SonarQube to behave unpredictably and data will be corrupted. There is no safeguard until [SONAR-10362](https://jira.sonarsource.com/browse/SONAR-10362).
 
-## Advanced configuration
+Setup a database by following the "Installing the Database" section of https://docs.sonarqube.org/latest/setup/install-server/.
 
-### Database configuration
+### Use bind-mounted folders
 
-By default, the image will use an embedded H2 database that is not suited for production.
+The images contain the SonarQube installation at `/opt/sonarqube`. You need to use bind-mounted folders to override selected files or directories :
 
-> Warning: Only a single instance of SonarQube can connect to a database schema. If you're using a Docker Swarm or Kubernetes, make sure that multiple SonarQube instances are never running on the same database schema simultaneously. This will cause SonarQube to behave unpredictably and data will be corrupted. There is no safeguard until [SONAR-10362](https://jira.sonarsource.com/browse/SONAR-10362).
+-	`/opt/sonarqube/conf`: configuration files, such as `sonar.properties`
+-	`/opt/sonarqube/data`: data files, such as the embedded H2 database and Elasticsearch indexes
+-	`/opt/sonarqube/logs`: contains SonarQube logs about access, web process, CE process, Elasticsearch logs
+-	`/opt/sonarqube/extensions`: plugins, such as language analyzers
+
+### First installation
+
+Follow these steps for your first installation:
+
+1. Create a `sonarqube_home` folder and create a var env `$SQ_HOME` pointing to it:
+```console
+$ mkdir /path/to/your/filesystem/sq_home
+$ export SQ_HOME=/path/to/your/filesystem/sq_home
+```
+2. Initialize SQ_HOME folder tree with `--init`. This will initialize the default configuration, copy embedded plugins, and prepare the data folder:
+```console
+$ docker run --rm \
+ -v $SQ_HOME/conf:/opt/sonarqube/conf \
+ -v $SQ_HOME/extensions:/opt/sonarqub
+ e/extensions \
+ -v $SQ_HOME/data:/opt/sonarqube/data \
+ sonarqube --init
+```
+3. Configure sonar.properties to configure the database settings. Templates are available for every supported database. Just uncomment and configure the template you need and comment out the lines dedicated to H2:
+```
+Example for PostgreSQL
+sonar.jdbc.username=sonarqube
+sonar.jdbc.password=mypassword
+sonar.jdbc.url=jdbc:postgresql://localhost/sonarqube
+```
+
+4. Drivers for the supported databases (except Oracle) are already provided. Do not replace the provided drivers; they are the only ones supported.
```

Comment:
I tried to add in the steps from the docs that related to this installation process. I wasn't really sure how important step 4 and step 5 are. Step 5, in particular, seems like something that could be more of a note, but I wasn't sure. Is this something that is really important for first installation and the user needs to do this in the process or can it be dropped from the actual installation process?

Reply:
activating the `-server` option is a real thing to be done. But changing the Java JVM point is not relevant, as the JRE is provided in the docker image and can't be changed. So you can remove this part

Reply:
I believe this advice on usage of `-server` is hardly relevant nowadays:

1. if it's on by default, so, for very largely most people, it's just noise
2. anyway, `-server` is on and can't be disabled on 64 bits env [since Java 7](https://docs.oracle.com/javase/7/docs/technotes/tools/windows/java.html) and we only support 64 bits env since SQ 7.7 (see [SONAR-11804](https://jira.sonarsource.com/browse/SONAR-11804)

Reply:
> if it's on by default

With this I was referring to (and assuming @pierre-guillot-sonarsource comment came from there) comments which says that in sonar.properties ([here](https://github.com/SonarSource/sonar-enterprise/blob/master/sonar-application/src/main/assembly/conf/sonar.properties#L84) and [there](https://github.com/SonarSource/sonar-enterprise/blob/master/sonar-application/src/main/assembly/conf/sonar.properties#L238))